models/issue_tracked_time.go

```diff
@@ -131,3 +225,105 @@ func TotalTimes(options FindTrackedTimesOptions) (map[*User]string, error) {
 	}
 	return totalTimes, nil
 }
+
+// DeleteIssueUserTimes deletes times for issue
+func DeleteIssueUserTimes(issue *Issue, user *User) error {
+	sess := x.NewSession()
+	defer sess.Close()
+
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	opts := FindTrackedTimesOptions{
+		IssueID: issue.ID,
+		UserID:  user.ID,
+	}
+
+	removedTime, err := deleteTimes(sess, opts)
+	if err != nil {
+		return err
+	}
+	if removedTime == 0 {
+		return ErrNotExist{}
+	}
+
+	if err := issue.loadRepo(sess); err != nil {
+		return err
+	}
+	if _, err := createComment(sess, &CreateCommentOptions{
+		Issue:   issue,
+		Repo:    issue.Repo,
+		Doer:    user,
+		Content: "- " + SecToTime(removedTime),
+		Type:    CommentTypeDeleteTimeManual,
+	}); err != nil {
+		return err
+	}
+
+	return sess.Commit()
+}
+
+// DeleteTime delete a specific Time
+func DeleteTime(t *TrackedTime) error {
+	sess := x.NewSession()
+	defer sess.Close()
+
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	if err := deleteTime(sess, t); err != nil {
+		return err
+	}
+
+	if _, err := createComment(sess, &CreateCommentOptions{
+		Issue:   t.Issue,
+		Repo:    t.Issue.Repo,
+		Doer:    t.User,
+		Content: "- " + SecToTime(t.Time),
+		Type:    CommentTypeDeleteTimeManual,
+	}); err != nil {
+		return err
+	}
+
+	return sess.Commit()
+}
+
+func deleteTimes(e Engine, opts FindTrackedTimesOptions) (removedTime int64, err error) {
```

Comment:
This function still feels remarkably inefficient.
Why are we extracting all of this data from the database to then simply remove it?

Wouldn't something like this work?
```go
var t *TrackedTime
err := options.ToSession(e).Select("count(time)").ForUpdate().Get(t)
if err != nil {
  return
}
removedTime = t.Time
err = options.ToSession(e).Delete(&TrackedTime{})
if err != nil {
   removedTime = 0
}
return
```

Reply:
This wont work ...

But i can make two querys:
 - sumint query for time
 - and a update statement wich set colum deleted to true

Reply:
sorry, yeah update rather than delete:

```go
err = options.ToSession(e).SetExpr("deleted", true).Update(&TrackedTime{})
```

Reply:
updated @zeripath but it doesnt work :(